client/my-sites/sites/sites.jsx

```diff
@@ -93,4 +108,13 @@ export default React.createClass( {
 			</Main>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
+		selectedSite: getSelectedSite( state ),
+		isPrivateSite: ( site ) => isPrivateSite( state, site.ID ),
```

Comment:
I would use `connect` to do it to have direct access to site prop. Plus you would easily get all 3 values mapping state to props: `isPrivateSite`, `isJetpackSite` and `isSiteUpgradeable`.